src/mozanalysis/config.py

```diff
@@ -17,7 +16,7 @@ class _ConfigLoader:
     Config objects are converted into mozanalysis native types.
     """
 
-    config_collection: Optional[ConfigCollection] = None
+    config_collection: ConfigCollection | None = None
```

Comment:
Tbh I find this and the other union type changes introduced in PEP 604 harder to read but I probably just need to get with the times. 

However, it looks like this rule, [UP007](https://docs.astral.sh/ruff/rules/non-pep604-annotation/), is marked unsafe for projects using runtime type annotations like Pydantic. Pydantic is in mozanalysis's dependency chain (metric-config-parser > mozilla-nimbus-schemas > pydantic). Is this rule safe for us to use?

Reply:
If I'm reading the documentation right it sounds likes it's only a concern before Python 3.10, which is our minimum version for the library.